flink-connector-elasticsearch6/src/main/java/org/apache/flink/streaming/connectors/elasticsearch/table/Elasticsearch6DynamicSource.java

```diff
@@ -0,0 +1,135 @@
+            @Nullable LookupCache lookupCache) {
+        this.format = format;
+        this.config = config;
+        this.physicalRowDataType = physicalRowDataType;
+        this.lookupMaxRetryTimes = lookupMaxRetryTimes;
+        this.lookupCache = lookupCache;
+    }
+
+    @Override
+    public LookupRuntimeProvider getLookupRuntimeProvider(LookupContext context) {
+        RestClientFactory restClientFactory = null;
+        if (config.getUsername().isPresent()
+                && config.getPassword().isPresent()
+                && !StringUtils.isNullOrWhitespaceOnly(config.getUsername().get())
+                && !StringUtils.isNullOrWhitespaceOnly(config.getPassword().get())) {
+            restClientFactory =
+                    new Elasticsearch6DynamicSink.AuthRestClientFactory(
+                            config.getPathPrefix().orElse(null),
+                            config.getUsername().get(),
+                            config.getPassword().get());
+        } else {
+            restClientFactory =
+                    new Elasticsearch6DynamicSink.DefaultRestClientFactory(
+                            config.getPathPrefix().orElse(null));
+        }
+
+        Elasticsearch6ApiCallBridge elasticsearch6ApiCallBridge =
+                new Elasticsearch6ApiCallBridge(config.getHosts(), restClientFactory);
+
+        // Elasticsearch only support non-nested look up keys
+        String[] keyNames = new String[context.getKeys().length];
+        for (int i = 0; i < keyNames.length; i++) {
+            int[] innerKeyArr = context.getKeys()[i];
+            Preconditions.checkArgument(
+                    innerKeyArr.length == 1, "Elasticsearch only support non-nested look up keys");
+            keyNames[i] = DataType.getFieldNames(physicalRowDataType).get(innerKeyArr[0]);
+        }
+
+        ElasticsearchRowDataLookupFunction<RestHighLevelClient> lookupFunction =
+                new ElasticsearchRowDataLookupFunction<>(
+                        this.format.createRuntimeDecoder(context, physicalRowDataType),
+                        lookupMaxRetryTimes,
+                        config.getIndex(),
+                        config.getDocumentType(),
+                        DataType.getFieldNames(physicalRowDataType).toArray(new String[0]),
+                        DataType.getFieldDataTypes(physicalRowDataType).toArray(new DataType[0]),
+                        keyNames,
+                        elasticsearch6ApiCallBridge);
+        if (lookupCache != null) {
+            return PartialCachingLookupProvider.of(lookupFunction, lookupCache);
+        } else {
+            return LookupFunctionProvider.of(lookupFunction);
+        }
+    }
+
+    @Override
+    public DynamicTableSource copy() {
+        return new Elasticsearch6DynamicSource(
+                format, config, physicalRowDataType, lookupMaxRetryTimes, lookupCache);
+    }
+
+    @Override
+    public String asSummaryString() {
+        return "Elasticsearch-6";
```

Comment:
```suggestion
        return "Elasticsearch6";
```
There is DynamicSink with `Elasticsearch6`, shouldn't we follow same naming conventions?

Reply:
wise consideration 👍